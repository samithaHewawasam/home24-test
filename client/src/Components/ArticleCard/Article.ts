export const article = {
  name: 'Beistelltisch Hoya (2er-Set)',
  index: 1,
  prices: {
    currency: 'EUR',
    regular: {
      value: 11499
    }
  },
  images: [
    {
      path: 'https://cdn1.home24.net/images/media/catalog/product/200x200/png/-/1/-1000174115-190524-15511900114-IMAGE-P000000001000174115.webp'
    }
  ]
}
