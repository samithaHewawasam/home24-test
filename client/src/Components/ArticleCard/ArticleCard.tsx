import { ShoppingCartOutlined } from '@ant-design/icons'
import { Card, Col, Image, Statistic } from 'antd'
import { Article } from 'Interfaces'
import React, { memo } from 'react'
import { formatter } from 'Utility'

interface IArticleCard {
  article: Article
  handleCartUpdate: any
}

const ArticleCard: React.FunctionComponent<IArticleCard> = memo(function ({
  article,
  handleCartUpdate
}: IArticleCard) {
  return (
    <Col
      xxl={4}
      xl={5}
      lg={6}
      md={6}
      sm={24}
      xs={24}
      style={{ textAlign: 'center' }}
    >
      <Card
        title={article.name}
        hoverable
        actions={[
          <ShoppingCartOutlined
            key={'ShopingCart'}
            onClick={() => {
              handleCartUpdate(article)
            }}
          />
        ]}
      >
        <Image
          alt={article.name}
          data-testid='article-image'
          src={article.images[0].path}
          style={{ marginTop: 5 }}
        />
        <Statistic
          value={formatter.format(article.prices.regular.value / 100)}
          precision={2}
        />
      </Card>
    </Col>
  )
})

ArticleCard.displayName = 'ArticleCard'
export default ArticleCard
