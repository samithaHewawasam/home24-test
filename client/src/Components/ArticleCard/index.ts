export { default as ArticleCard } from './ArticleCard'
export * from './Article'
