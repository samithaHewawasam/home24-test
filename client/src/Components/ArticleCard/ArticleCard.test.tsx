import { cleanup, render } from '@testing-library/react'
import { act } from '@testing-library/react-hooks/dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { article } from './Article'
import ArticleCard from './ArticleCard'

describe('Article Card', () => {
  const ArticleComponent = (
    <ArticleCard article={article} handleCartUpdate={jest.fn} />
  )

  /**
   * @description Render Component Promise
   */

  beforeEach(async () => {
    await act(async () => {
      render(ArticleComponent)
    })
  })

  /**
   * @description Clean the test case
   */

  afterEach(() => {
    cleanup()
  })

  /**
   * @description SnapShot for ArticleCard
   */
  it('renders ArticleCard correctly', () => {
    const tree = renderer.create(ArticleComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })

  /**
   * @description ArticleCard Component title should contain given text
   */

  it('render the Article Title', () => {
    const _ArticleCardTitle = document.querySelector(
      '.ant-card-head-title'
    )?.innerHTML
    expect(_ArticleCardTitle).toContain('Beistelltisch Hoya (2er-Set)')
  })

  /**
   * @description ArticleCard Component product image src should match
   */

  it('render the Article Product Image', () => {
    const _ProductImage = document.querySelector(
      '[data-testid=article-image] img'
    ) as HTMLImageElement
    expect(_ProductImage.src).toContain(article.images[0].path)
  })

  /**
   * @description ArticleCard Component product price should format
   */

  it('render the ArticleCard Price', () => {
    const _ArticleCardPrice = document.querySelector(
      '.ant-statistic-content-value'
    )?.innerHTML
    expect(_ArticleCardPrice).toContain('114,99&nbsp;â‚¬')
  })
})
