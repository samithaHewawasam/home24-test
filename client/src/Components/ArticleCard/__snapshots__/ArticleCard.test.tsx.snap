// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Card renders ArticleCard correctly 1`] = `
<div
  className="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-6 ant-col-lg-6 ant-col-xl-5 ant-col-xxl-4"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="ant-card ant-card-bordered ant-card-hoverable"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Beistelltisch Hoya (2er-Set)
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-image"
        data-testid="article-image"
        onClick={[Function]}
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <img
          alt="Beistelltisch Hoya (2er-Set)"
          className="ant-image-img"
          onError={[Function]}
          onLoad={[Function]}
          src="https://cdn1.home24.net/images/media/catalog/product/200x200/png/-/1/-1000174115-190524-15511900114-IMAGE-P000000001000174115.webp"
          style={
            Object {
              "height": undefined,
              "marginTop": 5,
            }
          }
        />
        <div
          className="ant-image-mask"
          style={
            Object {
              "display": undefined,
            }
          }
        >
          <div
            className="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              className="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
      <div
        className="ant-statistic"
      >
        <div
          className="ant-statistic-content"
        >
          <span
            className="ant-statistic-content-value"
          >
            114,99 €
          </span>
        </div>
      </div>
    </div>
    <ul
      className="ant-card-actions"
    >
      <li
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <span>
          <span
            aria-label="shopping-cart"
            className="anticon anticon-shopping-cart"
            onClick={[Function]}
            role="img"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              data-icon="shopping-cart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 00-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 100 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 00-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 00-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6z"
              />
            </svg>
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
